openwisp_firmware_upgrader/tests/test_api.py

```diff
@@ -817,23 +821,34 @@ def test_firmware_unauthorized(self):
                 r = client.get(url)
             self.assertEqual(r.status_code, 401)
 
+        # The download URL is handled by private_storage view which returns 403 for unauthorized users
         url = reverse('upgrader:api_firmware_download', args=[image.build.pk, image.pk])
         with self.subTest(url=url):
             with self.assertNumQueries(1):
                 r = client.get(url)
-            self.assertEqual(r.status_code, 401)
+            self.assertEqual(r.status_code, 403)
 
     def test_firmware_list(self):
         image = self._create_firmware_image()
-        self._create_firmware_image(type=self.TPLINK_4300_IL_IMAGE)
```

Comment:
Why did we remove this?

Reply:
I was debugging something
I missed to restore it before pushing the code
my bad